code/modules/events/holiday/vday.dm

```diff
@@ -59,23 +69,80 @@
 		if(!user.is_literate())
 			to_chat(user, "<span class='notice'>You scribble illegibly on [src]!</span>")
 			return
-		var/recipient = stripped_input(user, "Who is receiving this valentine?", "To:", null , 20)
-		var/sender = stripped_input(user, "Who is sending this valentine?", "From:", null , 20)
-		if(!user.canUseTopic(src, BE_CLOSE))
+		//Alright lets see who we are sending this bad boy too.
+		//Gets all the people on the z-level, don't want people meta dating nukies *too* hard.
+		//Also they only get one chance.
+		if(alert(user, "Are you sure you are ready to write your message? You only have one shot!", "Valentines message", "Yes!", "No...") == "No...")
+			to_chat(user, "<span class='notice'>You put down the pen thinking about who you want to send the message to.</span>")
+			return
+		var/turf/user_turf = get_turf(user)
+		if(!SSmobs.clients_by_zlevel[user_turf.z])
+			to_chat(user, "<span class='warning'>You stop and look around for a moment. Where the hell are you?</span>")
+			return
+		//No going back now
+		var/list/clients_on_level = SSmobs.clients_by_zlevel[user_turf.z]
+		var/list/mob_names = list()
+		for(var/mob/living/carbon/human/H in clients_on_level)
+			if(!H.mind || H == user)
+				//Ignore non-humans, they will be handled by the event.
+				continue
+			mob_names["[H.real_name]"] = H
+		if(!LAZYLEN(mob_names))
+			to_chat(user, "<span class='warning'>You feel empty and alone.</span>")
+			return
+		//Pick names
+		//At this point the user is shown the names of people on the z-level
+		//To prevent metastrats, you can only use this one time.
+		var/picked_name = input(user, "Who are you sending it to?", "Valentines Card", null) as null|anything in mob_names
+		var/mob/living/carbon/human/picked_human = mob_names[picked_name]
+		if(!picked_human || !istype(picked_human))
+			to_chat(user, "<span class='notice'>The card vanishes out of your hand! Lets hope they got it...</span>")
+			//rip
+			qdel(src)
+			return
+		if(!islist(GLOB.valentine_mobs))
+			to_chat(user, "<span class='warning'>You feel regret... It's too late now.</span>")
+			used = TRUE
+			return
+		if(used)
+			to_chat(user, "<span class='warning'>The card has already been used!</span>")
 			return
-		if(recipient && sender)
-			name = "valentine - To: [recipient] From: [sender]"
+		to_chat(user, "<span class='notice'>The card vanishes out of your hand! Lets hope they got it...</span>")
+		//List checking
+		GLOB.valentine_mobs[sender] = target
+		if(GLOB.valentine_mobs[target] == sender)
+			//wow.
+			forge_valentines_objective(sender, target)
+		//Off it goes!
+		//Create a new card to prevent exploiting
+		var/obj/item/valentine/new_card = new(get_turf(picked_human))
+		new_card.message = message
+		new_card.sender = user
+		new_card.target = picked_human
+		new_card.name = "valentines card from [new_card.sender]"
+		new_card.desc = "A Valentine's card! It is addressed to [new_card.target]."
```

Comment:
Should the new card have `used` set to true?

Reply:
oh I thought I set that